### ISSUE TITLE
Socket resource not released when performing GET request with an unresolvable URL

### ISSUE THREAD

**Issue by vanhoeij** [http4s/http4s#2890]
It looks like the socket is not released after performing a GET request with an unresolvable URL. I am not sure whether it is related to http4s (0.20.11) or its dependencies.

The following example can be used to reproduce the issue.

```scala
  def retry[A](io: IO[A]) : IO[A] = io.handleErrorWith(_ => IO.sleep(FiniteDuration(1, TimeUnit.SECONDS)) *> retry(io))

  val action = BlazeClientBuilder[IO](global)
    .resource
    .use(_.expect[String]("https://unknown.url/"))

  retry(action)
      .as(ExitCode.Success)
      .unsafeRunSync()
```

**Comment by vanhoeij** [comment]
I think that it is related to: https://github.com/http4s/blaze/issues/339

**Comment by rossabaker** [comment]
The blaze fix has been present since blaze-0.14.9, which was added in 0.20.13.
